### PR TITLE
Add MCD RPM builds on fcos branch of MCO

### PR DESCRIPTION
**Pull request by LorbusChris**
Analogous to #5572 

but without the changes in `ci-operator/config/openshift/origin/openshift-origin-master.yaml` (see https://github.com/openshift/release/pull/5572/files#diff-cd17a5d207369e19a76d7a3dbe835d38)

### CODE REVIEW

File: ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-fcos.yaml [hunk 1]
```diff
@@ -15,13 +15,26 @@ images:
 promotion:
   name: "4.3"
   namespace: origin
+raw_steps:
+- rpm_image_injection_step:
+    from: src
+    to: mco-rpms
```
Comment (smarterclayton) [inline ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-fcos.yaml]: this should be `machine-config-operator-rpms

Comment (smarterclayton) [reply ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-fcos.yaml]: You need rpm_image_injection_step to create `oc-base-with-rpms`, then promotion needs to look like it does in `openshift-oc-master.yaml`

**Comment by smarterclayton** [comment]
See my comment on the previous commit

**Comment by LorbusChris** [comment]
please have another look!

File: ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-fcos.yaml [hunk 2]
```diff
@@ -13,15 +13,32 @@ images:
   from: base
   to: machine-config-operator
 promotion:
+  additional_images:
+    machine-config-operator-rpms: rpms
+  excluded_images:
+  - machine-config-operator-base-with-rpms
   name: "4.3"
-  namespace: origin
```
Comment (LorbusChris) [inline ci-operator/config/openshift/machine-config-operator/openshift-machine-config-operator-fcos.yaml]: I'm confused, should this be origin or ocp?

**Comment by LorbusChris** [comment]
I'm wondering whether we also need a test to trigger the `rpms` target here, like in https://github.com/openshift/release/blob/master/ci-operator/config/openshift/oc/openshift-oc-master.yaml#L97-L100

**Comment by smarterclayton** [comment]
It can’t hurt, and makes the failure more obvious on a PR

**Comment by LorbusChris** [comment]
added the test

**Comment by LorbusChris** [comment]
Blocked by: https://github.com/openshift/machine-config-operator/pull/1225

**Comment by smarterclayton** [comment]
/lgtm

Can you port this to all the other branches 4.2+?

**Pull request merged**